Makd.mak

```
@@ -651,11 +651,10 @@ d2conv: $O/d2conv.stamp
 
 $O/d2conv.stamp: $C
 	$Vfind $C -type f -regex '^.+\.d$$' > $@
-ifeq "$(shell d1to2fix --help | grep -- --input)" ""
+	@if d1to2fix --help | grep -- --input; then \
+	$(call exec, d1to2fix $(DIMPORTPATHS) --fatal --input=$@) \
+	@else \
```

Now, wait, this wont work. That's not how make works.

Each line will invoke a shell (of course lines with _line continuation_ `\` are considered only one line), and that shell command can optionally start with `@` to avoid printing the command. So this `@else` will be passed to the shell literally as `@else`.

Or did you actually tried it and worked as you expected?

---

> Or did you actually tried it and worked as you expected?

It did :D

---

```
$ make d2conv V=2
find /home/dicebot/sociomantic/ocean -type f -regex '^.+\.d$' > /home/dicebot/sociomantic/ocean/build/devel/tmp/d2conv.stamp
d1to2fix -I/home/dicebot/sociomantic/ocean/build/devel/include -I./src  -I./submodules/makd/src --fatal --input=/home/dicebot/sociomantic/ocean/build/devel/tmp/d2conv.stamp 
```

---

Something is very fishy about this, because you should also see the output of the `d1to2fix --help | grep -- --input` command. I'll test.

---

Without `V=1` I get `/bin/sh: 4: Syntax error: end of file unexpected (expecting "fi")`. This is definitely broken, the statement before `else` also must end with `;` when its in one line. I don't know because of what black magic is working for you, but this is definitely broken.

---

Just leave it as it was before and add the `2>/dev/null` to suppress errors if you want your clean verbose (but not **too** verbose) output :D

---

Done